clippy_lints/src/matches.rs

```diff
@@ -254,7 +255,18 @@ fn report_single_match_single_pattern(cx: &LateContext<'_, '_>, ex: &Expr, arms:
     } else {
         SINGLE_MATCH
     };
-    let els_str = els.map_or(String::new(), |els| format!(" else {}", expr_block(cx, els, None, "..")));
+    let els_str = els.map_or(String::new(), |els| {
+        if in_macro(els.span) {
+            " else { .. }".to_string()
```

Comment:
Could we maybe get a better suggestion with [`source_callsite()`](https://doc.rust-lang.org/nightly/nightly-rustc/syntax_pos/struct.Span.html#method.source_callsite)

Everything else LGTM.